File: 718.cpp
//
//  Solution.cpp
//  LeetCode
//

#include "Solution.hpp"
#include <iostream>
#include <vector>
#include <map>

class Solution {
public:
    int findLength(std::vector<int>& A, std::vector<int>& B) {
        int res = 0;
        std::vector<std::vector<int>> f(A.size() + 1, std::vector<int>(B.size() + 1, 0));
        
        for (size_t i = 1; i <= A.size(); i++) {
            for (size_t j = 1; j <= B.size(); j++) {
                if (A[i - 1] == B[j - 1]) {
                    f[i][j] = f[i - 1][j - 1] + 1;
                    res = std::max(res, f[i][j]);
                }
            }
        }
        return res;
    }
};

File: 718.java
import java.util.*;

class Solution {
    public int findLength(int[] A, int[] B) {
        int res = 0;
        int[][] f = new int[A.length + 1][B.length + 1];
        for (int i = 1; i <= A.length; i++) {
            for (int j = 1; j <= B.length; j++) {
                if (A[i - 1] == B[j - 1]) {
                    f[i][j] = f[i - 1][j - 1] + 1;
                    res = Math.max(res, f[i][j]);
                }
            }
        }
        return res;
    }
}
